var/spack/repos/builtin/packages/py-python-language-server/package.py

```diff
@@ -0,0 +1,36 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyPythonLanguageServer(PythonPackage):
+    """An implementation of the Language Server Protocol for Python"""
+
+    homepage = "https://github.com/palantir/python-language-server"
+    url      = "https://github.com/palantir/python-language-server/archive/0.15.1.tar.gz"
+
+    version('0.15.1', 'deee6fe6776c4f59547939d356004650')
+
+    depends_on('py-setuptools', type='build')
```

Comment:
Missing many required dependencies:
```python
    install_requires=[                                                          
        'configparser',                                                         
        'future>=0.14.0',                                                       
        'futures; python_version == "2.7"',                                     
        'jedi>=0.10',                                                           
        'json-rpc==1.10.8',                                                     
        'mccabe',                                                               
        'pycodestyle',                                                          
        'pydocstyle>=2.0.0',                                                    
        'pyflakes',                                                             
        'rope>=0.10.5',                                                         
        'yapf',                                                                 
        'pluggy'                                                                
    ],          
```

Reply:
thanks @adamjstewart , I added a few and fill in the rest later...

Reply:
They should all be `type=('build', 'run')`